compiler/src/dotty/tools/dotc/transform/Recheck.scala

```diff
@@ -212,7 +212,9 @@ abstract class Recheck extends Phase, SymTransformer:
       mt.instantiate(argTypes)
 
     def recheckApply(tree: Apply, pt: Type)(using Context): Type =
-      recheck(tree.fun).widen match
+      // reuse the tree's type if the function was a signature polymorphic method, instead of rechecking it
+      val tpe = if tree.fun.symbol.originalSignaturePolymorphic.exists then tree.fun.tpe else recheck(tree.fun)
```

Comment:
Shouldn't we at least `recheck(tree.fun.qual)`? It seems we're losing an entire subtree here.